x-pack/plugins/stack_alerts/public/alert_types/es_query/expression.tsx

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, Fragment, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import 'brace/theme/github';
+import { XJsonMode } from '@kbn/ace';
+
+import {
+  EuiButtonEmpty,
+  EuiCodeEditor,
+  EuiSpacer,
+  EuiFormRow,
+  EuiCallOut,
+  EuiText,
+  EuiTitle,
+  EuiLink,
+} from '@elastic/eui';
+import { DocLinksStart, HttpSetup } from 'kibana/public';
+import { XJson } from '../../../../../../src/plugins/es_ui_shared/public';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import {
+  getFields,
+  COMPARATORS,
+  ThresholdExpression,
+  ForLastExpression,
+  AlertTypeParamsExpressionProps,
+} from '../../../../triggers_actions_ui/public';
+import { validateExpression } from './validation';
+import { parseDuration } from '../../../../alerts/common';
+import { buildSortedEventsQuery } from '../../../common/build_sorted_events_query';
+import { EsQueryAlertParams } from './types';
+import { IndexSelectPopover } from '../components/index_select_popover';
+
+const DEFAULT_VALUES = {
+  THRESHOLD_COMPARATOR: COMPARATORS.GREATER_THAN,
+  QUERY: `{
+  "query":{
+    "match_all" : {}
+  }
+}`,
+  TIME_WINDOW_SIZE: 5,
+  TIME_WINDOW_UNIT: 'm',
+  THRESHOLD: [1000],
+};
+
+const expressionFieldsWithValidation = [
+  'index',
+  'esQuery',
+  'timeField',
+  'threshold0',
+  'threshold1',
+  'timeWindowSize',
+];
+
+const { useXJsonMode } = XJson;
+const xJsonMode = new XJsonMode();
+
+interface KibanaDeps {
+  http: HttpSetup;
+  docLinks: DocLinksStart;
+}
+
+export const EsQueryAlertTypeExpression: React.FunctionComponent<
```

Comment:
there's a lot going on in this component - can we add some unit tests around it?

Reply:
Added unit test [here](https://github.com/elastic/kibana/pull/88528/commits/0b6a8d8baf895928f542c11a483e2bcc16de8a76)